### PR TITLE
Ide analyzer options

### PR DESCRIPTION
**Pull request by tmat**
Moves options that are not defined in editorconfig but used in analyzers to `IdeOptions` accessible from `WorkspaceAnalyzerOptions`.

Makes `SymbolSearchOptions` global.

Depends on https://github.com/dotnet/roslyn/pull/58747

### CODE REVIEW

File: src/Features/Core/Portable/Diagnostics/WorkspaceAnalyzerOptions.cs [hunk 1]
```diff
@@ -20,12 +21,27 @@ internal sealed class WorkspaceAnalyzerOptions : AnalyzerOptions
...
+            _ideOptionsCache = ImmutableDictionary<string, IdeAnalyzerOptions>.Empty;
         }
 
+        public IdeAnalyzerOptions GetIdeOptions(string language)
+            => ImmutableInterlocked.GetOrAdd(
+                ref _ideOptionsCache,
+                language,
+                static (language, solution) => new IdeAnalyzerOptions(
+                    FadeOutUnusedImports: solution.Options.GetOption(Fading.FadingOptions.Metadata.FadeOutUnusedImports, language),
+                    FadeOutUnreachableCode: solution.Options.GetOption(Fading.FadingOptions.Metadata.FadeOutUnusedImports, language),
```
Comment (JoeRobich) [inline src/Features/Core/Portable/Diagnostics/WorkspaceAnalyzerOptions.cs:40]:
```suggestion
                    FadeOutUnreachableCode: solution.Options.GetOption(Fading.FadingOptions.Metadata.FadeOutUnreachableCode, language),
```

Comment (tmat) [reply src/Features/Core/Portable/Diagnostics/WorkspaceAnalyzerOptions.cs:40]: Oops, good catch!

File: src/Features/Core/Portable/AddImport/AbstractAddImportCodeFixProvider.cs [hunk 2]
```diff
@@ -60,21 +60,19 @@ public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
...
             var placement = await AddImportPlacementOptions.FromDocumentAsync(document, cancellationToken).ConfigureAwait(false);
 
             var options = new AddImportOptions(
-                context.Options.SearchReferenceAssemblies,
+                context.Options.SearchOptions,
                 context.Options.HideAdvancedMembers,
                 placement);
 
-            var symbolSearchService = options.SearchReferenceAssemblies || searchNuGetPackages
+            var symbolSearchService = options.SearchOptions.SearchReferenceAssemblies
```
Comment (JoeRobich) [inline src/Features/Core/Portable/AddImport/AbstractAddImportCodeFixProvider.cs:70]: This looks like a behavior change that impacts getting the `packageSources` below.

Comment (tmat) [reply src/Features/Core/Portable/AddImport/AbstractAddImportCodeFixProvider.cs:70]: Will fix.

**Review by JoeRobich — APPROVED** [review]

**Pull request merged**